lib/facterdb.rb

```diff
@@ -4,7 +4,35 @@
 module FacterDB
 
   def self.database
-    @database ||= "[#{Dir.glob("#{File.expand_path(File.join(File.dirname(__FILE__), '../facts'))}/*/*.facts").map { |f| File.read(f) }.join(',')}]\n"
+    @database ||= "[#{facterdb_fact_files.map { |f| File.read(f) }.join(',')}]\n"
+  end
+
+  # @return [Array[String]] -  list of all files found in the default facterdb facts path
+  def self.default_fact_files
+    proj_root = File.join(File.dirname(File.dirname(__FILE__)))
+    facts_dir = File.expand_path(File.join(proj_root, 'facts'))
+    Dir.glob(File.join(facts_dir, "**", '*.facts'))
+  end
+
+  # @return [Array[String]] -  list of all files found in the user supplied facterdb facts path
+  # @param fact_paths [String] - a comma separated list of paths to search for fact files
+  def self.external_fact_files(fact_paths = ENV['FACTERDB_SEARCH_PATH'])
+    fact_paths ||= ''
+    return [] if fact_paths.empty?
+    paths = fact_paths.split(',').map do |fact_path|
+      unless File.directory?(fact_path)
+        warn("[FACTERDB] Ignoring external facts path #{fact_path} as it is not a directory")
+        next nil
+      end
+      File.join(fact_path.strip, '**', '*.facts')
+    end.compact
+    Dir.glob(paths)
```

Comment:
`Dir.glob` behaves badly when passed Windows style backslash delimited paths, so I'd recommend adding the following line to the above map before the final `File.join` to convert the backslashes.

```ruby
fact_path = fact_path.gsub(File::ALT_SEPARATOR, File::SEPARATOR) if File::ALT_SEPARATOR
```

Reply:
These are great suggestions.  Fixed in latest push.